packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -91,139 +123,600 @@ const propTypes = {
+        inputValue = `0${inputValue}`;
+      }
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.DAY);
+  };
+
+  const handleMonthChange = (event) => {
+    let inputValue = event.target.value;
+    if (!DateUtil.validDateInput(inputValue)) {
+      return;
+    }
+
+    // Ignore the entry if the value did not change or it is invalid.
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 2.
+    if (inputValue === date.month || inputValue.length > 2 || Number(inputValue) > 12 || inputValue === '00') {
+      return;
+    }
+
+    // If the change made was not a deletion of a digit, then prepend '0'
+    // when the input value is a single digit value between 2 and 9
+    if (inputValue.length >= date.month.length) {
+      const digitsToPrependZero = ['2', '3', '4', '5', '6', '7', '8', '9'];
+
+      if (digitsToPrependZero.indexOf(inputValue) > -1) {
+        inputValue = `0${inputValue}`;
       }
-    } else if (inputAttributes && inputAttributes['aria-describedby']) {
-      ariaDescriptionIds = inputAttributes['aria-describedby'];
     }
 
-    const format = intl.formatMessage({ id: 'Terra.datePicker.dateFormat' });
+    handleDateChange(event, inputValue, DateUtil.inputType.MONTH);
+  };
+
+  const handleYearChange = (event) => {
+    const inputValue = event.target.value;
+    if (!DateUtil.validDateInput(inputValue)) {
+      return;
+    }
+
+    // Ignore the entry if the value did not change or it is invalid.
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 4.
+    if (inputValue === date.year || inputValue.length > 4) {
+      return;
+    }
+
+    // Ignore the 3rd entry if the first two digits are not 19, 20 or 21
+    if (inputValue.length === 3 && (Number(inputValue) < 190 || Number(inputValue) > 210)) {
+      return;
+    }
+
+    // Ignore the 4th entry if the year value is not between MIN_YEAR and MAX_YEAR
+    if (inputValue.length === 4 && (Number(inputValue) < Number(DateUtil.MIN_YEAR) || Number(inputValue) > Number(DateUtil.MAX_YEAR))) {
+      return;
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.YEAR);
+  };
+
+  const setInputFocus = (event, inputRef, selectionStart, selectionEnd) => {
+    inputRef.focus();
+    inputRef.setSelectionRange(selectionStart, selectionEnd);
+    event.preventDefault();
+  };
+
+  const handleDayInputKeydown = (event) => {
+    if (inputAttributes?.readOnly) {
+      return;
+    }
 
-    return (
-      <div className={cx(theme.className)}>
-        <div className={cx('custom-input')}>
+    if (event.keyCode === KEY_UP) {
+      const incrementedDay = DateUtil.incrementDay(date.day, date.month, date.year);
+      handleDateChange(event, incrementedDay, DateUtil.inputType.DAY);
```

Comment:
Before calling this handle change event for Day, Month and Year. can we have a check to validate whether the current value in the input is Max or Min value allowed for the field ? (for ex :-  2100 is the max year input and 1900 is min year input). with the current code we will be calling the onChange 
event even though if the current year input value is 2100 and in 'KEY UP' event there is no change in the year input value but onchange is still called.

Reply:
Prevented onChange from being triggered for the edge cases where incremented or decremented value is the same as the current value : https://github.com/cerner/terra-framework/pull/1208/commits/c4a158635cefd04df00c2ce4a8fb62f70bc43fc9